### PR TITLE
test: minor e2e test enhancements

### PR DESCRIPTION
**Pull request by mxschmitt**
Hey! Playwright maintainer here, happy to see that you're using Playwright. Just a few minor changes about how you use Playwright.

### CODE REVIEW

File: tests/auth.test.ts [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
-    expect(page.getByText("Signed In")).not.toBeVisible();
-    expect(page.getByText("Signed Out")).toBeVisible();
+    await expect(page.getByText("Signed Out")).toBeVisible();
+    await expect(page.getByText("Signed In")).toBeHidden();
   });
 
   test("User can sign in and out", async () => {
-    await page.click("text=Sign In");
-    await page.waitForSelector("text=Sign Out");
+    await page.getByRole("button", { name: "Sign In" }).click();
```
Comment (mxschmitt) [inline tests/auth.test.ts:19]: We recommend using role selectors, which do access-ability testing as a drive-by. All our assertions wait automatically, so you don't need waitForSelector. By using expect, you'll also get good error reporting.

File: tests/firestore.test.ts [hunk 2]
```diff
@@ -1,23 +1,21 @@
...
-	expect( (await page.$$('li')).length ).toBe(2);
-	expect(await page.textContent('li')).toContain('firestore item');
-    
+	await page.getByRole('button', { 'name': 'Sign In'}).click();
+	await expect(page.getByTestId('count')).toContainText('0 posts');
+	await page.getByRole('button', { name: 'Add Post' }).click();
+	await expect(page.getByTestId('count')).toContainText('1 posts');
+	await page.getByRole('button', { name: 'Add Post' }).click();
+	await expect(page.getByTestId('count')).toContainText('2 posts');
+	await expect(page.locator('li')).toHaveCount(2);
```
Comment (mxschmitt) [inline tests/firestore.test.ts:16]: General rule, never do manual sync expect's, Playwright has web-first assertions which will retry over and over again until a condition is met or the timeout is reached, Your manual toBe length comparison won't do that. See here: https://playwright.dev/docs/test-assertions

File: tests/main.test.ts [hunk 3]
```diff
@@ -2,11 +2,11 @@ import { expect, test } from '@playwright/test';
 
 test('SvelteFire app initializes properly', async ({ page }) => {
 	await page.goto('/');
-	expect(await page.textContent('h1')).toBe('Welcome to SvelteFire');
+	await expect(page.locator('h1')).toHaveText('Welcome to SvelteFire');
```
Comment (mxschmitt) [inline tests/main.test.ts:5]: This got changed from a manual expect to a Playwright web-first assertions expect.

File: tests/auth.test.ts [hunk 4]
```diff
@@ -1,24 +1,27 @@
...
   test.beforeAll(async ({ browser }) => {
     page = await browser.newPage();
     await page.goto("/auth-test");
   });
+  test.afterAll(async () => {
+    await page.close();
+  });
 
   test("Renders UI conditionally based on auth state", async () => {
-    expect(page.getByText("Signed In")).not.toBeVisible();
```
Comment (mxschmitt) [inline tests/auth.test.ts:11]: Here were awaits missing, we recommend no-floating-promises for that: https://playwright.dev/docs/best-practices#lint-your-tests

**Comment by codediodeio** [comment]
Thank you Max!
